### PR TITLE
analyzers: default A/B-tier language coverage

### PR DESCRIPTION
**Pull request by alexhawat**
## What?

Reviews now run language-default linters and scanners that used to stay opt-in, and pick up missing coverage for TypeScript, Python security, Go and Rust advisories, unused code, and copy-paste detection.

- Go, Rust, Ruby, and PHP now get a default lint path when the repo looks like those languages.
- Nine new tools cover TypeScript, Python security and dead code, Go/Rust advisories, unused JS exports, copy-paste, and typos.
- A-tier languages that had no default signal now have one, including Java, C/C++, Kotlin, Swift, SQL, CSS, HTML, YAML, Make, Markdown, and Terraform.
- B-tier languages get the same treatment: Lua, Fortran, Rego, PowerShell, Batch, Liquid, Smarty, Ember, and Prisma.

## Why?

Coverage sweep #309–#339. PRs in those stacks used to ship with an empty default analyzer set unless someone opted in by hand.

## Note

Analyzer coverage sweep #309–#339: 4 cheap flips (W), 9 new manifests (X), 19 A-tier (Y), 9 B-tier (Z).

Thermos quality + review were clean at `068300e3` (NO ISSUES). After rebase onto `origin/pre-0.0.1` that tip is `dcea2c50`.

Draft-close comments for #309–#339 were not posted to GitHub (D5). Issues stay open until a maintainer closes them.

## Test plan

- [x] `make lint`
- [x] `make typecheck`
- [x] `make catalog-check` (67 manifests)

## Related PR(s)/Issue(s)

Related: #309–#339

### CODE REVIEW

**Review by sevn-one — COMMENTED** [review]
## Approved

Reviewed the full diff (catalog flips, new manifests, native parsers, resolve/registry/adapters changes, and the ~2k lines of new tests). CI is green on `dcea2c5` and the sweep is well-covered: every auto-flip, config gate, parser shape, and diff-filter behavior has a real (non-xfail) assertion.

Key correctness checks that held up:
- `resolve_analyzer` ladder refactor is behavior-preserving; the sqlfluff no-dialect skip and the post-ladder config-absent patches (phpstan `--level=0`, prisma-lint fallback `--config`) apply only to plans with argv.
- `filter_lint_targets_for_manifest` correctly keeps enable markers (`composer.json`, `go.mod`, `.sqlfluff`, `pyproject.toml`) out of `{files}` expansion.
- D11 (rubocop config gate), D16 (mypy default), D17 (biome>eslint>oxlint by config presence), Rails/shopify/ember gates, and the sqlfluff dialect gate all behave as documented and are pinned by fixtures.
- New native parsers raise on non-diagnostic stdout (tool errors become skips, never false-clean), and `parse_sarif` via `require_json_object` still validates the version.

Non-blocking observations (no correctness/security regression, can follow up):
1. A PR that changes only an enable marker (e.g. `composer.json` or `go.mod`) passes `detect_enabled` but yields zero lint targets — phpstan/sqlfluff then run with no paths (ending in a confusing "failed to parse analyzer output" skip) and golangci-lint runs unscoped. `test_lint_target_globs` pins the empty-`[]` contract but nothing exercises the marker-only end-to-end path.
2. `parse_bundler_audit_json` anchors every finding to `Gemfile.lock:1`, and `parse_cargo_deny_json` falls back to `Cargo.toml:1` when a diagnostic carries no file label — acceptable, low-fidelity anchors.

No changes requested.

---
*via mergecraft* · `nous/deepseek/deepseek-v4-flash` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32396301401)

**Comment by sevn-one** [comment]
Reviewed PR #376 (analyzers: default A/B-tier language coverage).

**Verdict: Approve** — no changes requested. Review published at https://github.com/alexhawat/mergeCraft/pull/376#pullrequestreview-4985517187.

**What was verified (full-diff review):**
- `resolve_analyzer` ladder refactor is behavior-preserving; sqlfluff no-dialect skip and post-ladder config-absent patches (phpstan `--level=0`, prisma-lint fallback `--config`) are correct.
- D11 (rubocop config gate), D16 (mypy default), D17 (biome>eslint>oxlint by config presence), Rails/shopify/ember gates, and sqlfluff dialect gate behave as documented and are pinned by fixtures.
- New native parsers raise on non-diagnostic stdout (tool errors → skips, never false-clean); `parse_sarif` via `require_json_object` still validates version.
- `filter_lint_targets_for_manifest` keeps enable markers out of `{files}` expansion.
- jscpd scratch-dir `--output` + output_path rewrite correct; bandit version matches pyproject pin.

**Non-blocking observations (in review body):** marker-only PRs (e.g. `composer.json`/`go.mod`) yield empty lint targets → phpstan/sqlfluff run with no args (confusing parse-skip), golangci-lint runs unscoped; bundler-audit/cargo-deny findings anchor to lockfile line 1.

**Analyzers pre-merge row: skipped** — trufflehog/mypy/ruff/semgrep all unavailable in this environment (sandbox/trusted-tier constraints); no analyzer findings.

Note: `submit_review_verdict(approve)` was rejected as a terminal-submission conflict for this run; the review itself is published with an explicit approve statement.

---
*via mergecraft* · `nous/deepseek/deepseek-v4-flash` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32396301401)

File: src/mergecraft/analyzers/catalog/bundler-audit.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+id: bundler-audit
+category: vuln
+languages:
+- ruby
+detect:
+  files:
+  - Gemfile.lock
+command:
+- bundle
```
Comment (sevn-one) [inline src/mergecraft/analyzers/catalog/bundler-audit.yaml]: _Functional Correctness_ | _Major_ | _Quick win_ | _certain_

This resolves and invokes Bundler itself as `bundle audit`, but `bundler-audit` 0.9.2 exposes the separate `bundle-audit check --format json` CLI; modern Bundler does not provide an `audit` subcommand. Every auto-enabled run on `Gemfile.lock` therefore becomes a parse skip, so switch to the actual executable/subcommand and add a finalized-argv regression test.

File: src/mergecraft/analyzers/catalog/tflint.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@ command:
 scope: diff
 parser: sarif
 supports_fix: false
-default_enabled: false
+default_enabled: auto
```
Comment (sevn-one) [inline src/mergecraft/analyzers/catalog/tflint.yaml:16]: _Functional Correctness_ | _Major_ | _Quick win_ | _certain_

Auto-enabling this manifest exposes its `{files}` expansion, but TFLint 0.54 accepts no positional file arguments—it lints the working directory and scopes with `--filter`/`--chdir`. Terraform changes therefore produce an invalid command and a skipped analyzer; make the finalized argv valid for TFLint and cover it with an argv-level test.

**Review by sevn-one — CHANGES_REQUESTED** [review]
Two newly auto-enabled analyzers still build invalid commands, so their advertised default coverage is skipped at runtime. The green suite covers detection and parser fixtures but not these finalized argv contracts.

---
*via mergecraft* · `openai/gpt-5.6-sol` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32396301401)

**Review by sevn-one — CHANGES_REQUESTED** [review]
The two argv-contract findings raised in the prior `CHANGES_REQUESTED` review remain unresolved at head `a421986`:

1. **`bundler-audit.yaml`** — command is `bundle audit --format json`. bundler-audit v0.9.2 ships the `bundle-audit` executable (default task `check`) and documents `bundle-audit check --format json`; it registers no Bundler `audit` subcommand. Every auto-enabled run on a `Gemfile.lock` repo fails with "Could not find command 'audit'" and turns into a parse skip, silently killing the advertised Ruby vuln coverage.
2. **`tflint.yaml`** — pins TFLint 0.54.0 but still expands `{files}` as positional args. TFLint removed CLI file arguments in v0.47 (they now return an error; usage is `tflint --chdir=DIR [OPTIONS]` with `--filter` for scoping), so auto-enabled Terraform changes produce an invalid command and a skipped analyzer.

Commit `a421986` fixes the earlier review's non-blocking observations (marker-only skip plans, dropped fake line-1 anchors) — verified via `test_lint_target_globs.py` and the `start_line: int | None` refactor — but does not touch either manifest. Please fix both finalized argv contracts and add argv-level regression tests.

---
*via mergecraft* · `nous/deepseek/deepseek-v4-flash` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32398524308)

**Comment by sevn-one** [comment]
Reviewed PR #376 (analyzers: default A/B-tier language coverage) — verdict: **request changes**.

**Blocking findings (both from the prior CHANGES_REQUESTED review, verified still present at head `a421986`):**
1. `bundler-audit.yaml` command is `bundle audit --format json`. bundler-audit v0.9.2 ships the `bundle-audit` executable (default task `check`; README documents `bundle-audit check --format json`) and registers no Bundler `audit` subcommand — every auto-enabled run on a `Gemfile.lock` repo fails and becomes a parse skip, killing the advertised Ruby vuln coverage.
2. `tflint.yaml` pins 0.54.0 but expands `{files}` positionally; TFLint removed CLI file arguments in v0.47 (they error), so auto-enabled Terraform changes produce an invalid command.

**What was verified / passes:** commit `a421986` correctly fixes the earlier non-blocking observations — marker-only PRs now skip via `filter_lint_targets_for_manifest` + `_NO_LINT_TARGETS_REASON` (pinned by `test_lint_target_globs.py`), and bundler-audit/cargo-deny findings drop fake line-1 anchors via the `start_line: int | None` refactor (pinned by `test_auto_enabled_native.py`). Native parsers are fail-closed on non-diagnostic stdout, `parse_sarif` still validates version, and config gates (D11 rubocop, D16 mypy, D17 biome>eslint>oxlint, sqlfluff dialect) behave as documented with real fixtures. No security regressions found; new findings parsed correctly with path-only anchors.

**Analyzers pre-merge row: skipped** — trufflehog/mypy/ruff/semgrep all unavailable in this environment (sandbox/trusted-tier constraints).

Review published: https://github.com/alexhawat/mergeCraft/pull/376#pullrequestreview-4985732538

---
*via mergecraft* · `nous/deepseek/deepseek-v4-flash` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32398524308)

**Review by sevn-one — CHANGES_REQUESTED** [review]
The two merge-blocking argv contracts from the existing threads remain unchanged at `a421986`: `bundler-audit` still invokes Bundler as `bundle audit`, and TFLint still receives changed files as positional arguments. The latest commit improves marker-only handling and finding anchors, but the added tests still cover detection/parser behavior rather than finalized executable commands; please fix both commands and add argv-level regression coverage.

---
*via mergecraft* · `openai/gpt-5.6-sol` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32398524308)

**Review by sevn-one — COMMENTED** [review]
## Approved

Well-structured sweep of issues #309–#339. Key observations:

**Soundness**
- Two prior review threads resolved in the current diff:
  - bundler-audit: command is now `bundler-audit check --format json` (not `bundle audit`), with a finalized-argv regression test at `test_b_tier_flips.py:6303`.
  - tflint: `{files}` removed; tool lints workspace, `scope: diff` filters findings. Verified by `test_a_tier_residuals.py:6620`.

**Design**
- `start_line` / `end_line` made nullable (`finding.py:60-63`) — necessary for bundle-level advisories and repo-scoped tools that don't report line numbers. Budget, scope, baseline suppression, SARIF export, and dedup all handle `None` correctly.
- Config-absent patches (`resolve.py:186-227`) for phpstan `--level=0` and prisma-lint fallback are clean and test-covered.
- Marker-only lint skip (`execution.py:64-67`) prevents tools from running empty `{files}` expansion when only `composer.json`/`go.mod`/`.sqlfluff` changes.
- js-linter intent rewritten to D17 config-file-first semantics (`detect.py:182-213`).

**Parsers**
- 17 new native parsers (JSONL, JSON, text-line). Every one has a fixture, a happy-path test, and a garbage-input test.
- SARIF parser refactored to use shared `require_json_object`.

**Testing**
- 4 new test files + amendments covering Batch W/X/Y/Z, lint-target globs, and native parser edge cases.
- No remaining xfails; 0 XPASS as documented in the test plan.

**Nit (non-blocking)**: `_line_intersects_hunks` at `scope.py:166` returns `True` when `start_line` is `None` — correct (conservative keep). Worth a comment for future readers.

---
*via mergecraft* · `nous/deepseek/deepseek-v4-flash` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32400508068)

**Comment by sevn-one** [comment]
## Review Complete

**Verdict**: Code changes are correct and well-tested. All prior review threads are resolved. No new issues found.

**Note**: The `approved: true` parameter was serialized as a string in the first API call, causing the review to post as `COMMENTED` instead of `APPROVED`. The substantive review content is accurate — no changes requested.

---
*via mergecraft* · `nous/deepseek/deepseek-v4-flash` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32400508068)

File: src/mergecraft/analyzers/catalog/bandit.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+languages:
+  - python
+detect:
+  files:
+    - "*.py"
+command:
+  - bandit
+  - -r
+  - --format
+  - sarif
```
Comment (sevn-one) [inline src/mergecraft/analyzers/catalog/bandit.yaml]: Bandit 1.9.4 exposes SARIF only through its optional `sarif` extra, while this repo pins `bandit[toml]` and repo-native users may have plain Bandit. There `--format sarif` is unavailable, so this auto-enabled analyzer skips instead of providing Python security coverage. Use built-in JSON with a native parser, or guarantee `bandit[sarif]` for every resolved toolchain, and cover the finalized command with a real CLI regression test.

**Review by sevn-one — CHANGES_REQUESTED** [review]
The two prior argv blockers are fixed at `c743b15`. One new analyzer still has an output-format dependency that the pinned/repo-native toolchain does not guarantee, so the advertised Python security coverage can be skipped.

---
*via mergecraft* · `openai/gpt-5.6-sol` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32400508068)

**Review by sevn-one — COMMENTED** [review]
Approved. The latest commit moves Bandit to built-in JSON, wires in the native parser, and adds finalized-argv coverage. The prior Bundler Audit and TFLint command issues are also fixed, so all merge-blocking threads are addressed.

---
*via mergecraft* · `openai/gpt-5.6-sol` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32402273444)
